locksmith/src/operations/stripeOperations.ts

```diff
@@ -59,7 +59,35 @@ export const saveStripeCustomerIdForAddress = async (
   }
 }
 
+/**
+ * Method which delets the stripe customer id for an address
+ */
+export const deletePaymentDetailsForAddress = async (
+  publicKey: ethereumAddress
+) => {
+  const normalizedEthereumAddress = Normalizer.ethereumAddress(publicKey)
+
+  // First, let's delete the StripeCustomer
+  const deletedStripeCustomer = await StripeCustomer.destroy({
+    where: { publicKey: { [Op.eq]: normalizedEthereumAddress } },
+  })
+
+  // Then, update UserReference
+  // TODO: deprecate when all stripe_customer_id UserReferences have been moved to use StripeCustomer
+  const [updatedUserReference] = await UserReference.update(
+    {
+      stripe_customer_id: '',
+    },
+    {
+      where: { publicKey: { [Op.eq]: normalizedEthereumAddress } },
+    }
+  )
+
+  return deletedStripeCustomer > 0 || updatedUserReference > 0
```

Comment:
What are some of the reasons that this would not be  true? It looks like `updatedUserReference` is initially stored into its own array, here it looks like we are checking to see if the result is greater than zero is this design intentional?

Reply:
Sequlize will return the number of rows affected. If this is 0 it means the row was not changed.

Reply:
The second object in the array is the list of rows affected.